library/init/data/set.lean

```diff
@@ -95,4 +91,22 @@ instance : functor set :=
                   λ ⟨b, ⟨⟨a, ⟨h₁, h₂⟩⟩, h₃⟩⟩, ⟨a, ⟨h₁, h₂.symm ▸ h₃⟩⟩⟩
  end}
 
+@[reducible]
+def Union {α : Sort u} (s : α → set β) : set β := {x | ∃ i, x ∈ s i}
+
+@[reducible]
+def Inter {α : Sort u} (s : α → set β) : set β := {x | ∀ i, x ∈ s i}
+
+notation `⋃` binders `, ` r:(scoped f, Union f) := r
+notation `⋂` binders `, ` r:(scoped f, Inter f) := r
+
+@[reducible]
+def sUnion (s : set (set α)) : set α := {t | ∃ a ∈ s, t ∈ a}
+
+@[reducible]
+def sInter (s : set (set α)) : set α := {t | ∀ a ∈ s, t ∈ a}
+
+prefix `⋃₀`:110 := sUnion
```

Comment:
At the very least, `Union` needs to be renamed to `sUnion` for consistency. The rest is just for uniformity. I already have the repair to library_dev prepared, but since notation can't be shadowed (it instead introduces ambiguous translations, which cause errors), I needed to prime the notation as well:

     def Union' {α : Sort u} (s : α → set β) : set β := supr s
     
     @[reducible]
    -def Inter {α : Sort u} (s : α → set β) : set β := infi s
    +def Inter' {α : Sort u} (s : α → set β) : set β := infi s
     
    -notation `⋃` binders `, ` r:(scoped f, Union' f) := r
    -notation `⋂` binders `, ` r:(scoped f, Inter f) := r
    +notation `⋃'` binders `, ` r:(scoped f, Union' f) := r
    +notation `⋂'` binders `, ` r:(scoped f, Inter' f) := r

To avoid confusion, I'll remove this and put it in a separate PR, though.